### PR TITLE
tkt-72081: fix(middlewared/crypto): make sure directory tree exists

### PR DESCRIPTION
**Pull request by william-gr**
Ticket:	#72081

### CODE REVIEW

File: src/middlewared/middlewared/plugins/crypto.py [hunk 1]
```diff
@@ -741,7 +741,7 @@ def __init__(self, *args, **kwargs):
             ))
 
         if not os.path.exists(root_path):
-            os.mkdir(root_path, 0o755)
+            os.makedirs(root_path, 0o755, exist_ok=True)
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/crypto.py]: `if not os.path.exists(root_path)` check has no sense anymore

Comment (william-gr) [reply src/middlewared/middlewared/plugins/crypto.py]: Not true, there could be a race condition.

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/crypto.py]: Agreed :)

Comment (themylogin) [reply src/middlewared/middlewared/plugins/crypto.py]: What race condition? `exist_ok=True`

Comment (william-gr) [reply src/middlewared/middlewared/plugins/crypto.py]: Yes, exist_ok=True handles the race condition and its the only reason its being used.

I prefer to check for the existence before blindly calling makedirs and raising an exception. That does not make os.path.exists useless, it avoids extra calls/checks and an exception handling.

**Pull request merged**